src/DynamoCore/DSEngine/EngineController.cs

```diff
@@ -51,9 +48,53 @@ private void OnTraceReconciliationComplete(TraceReconciliationEventArgs e)
         private readonly Queue<GraphSyncData> graphSyncDataQueue = new Queue<GraphSyncData>();
         private readonly Queue<List<Guid>> previewGraphQueue = new Queue<List<Guid>>();
         public bool VerboseLogging;
+
         private readonly Object macroMutex = new Object();
 
-        public static CompilationServices CompilationServices; 
+        public static CompilationServices CompilationServices;
+
+        /// <summary>
+        /// Get DesignScript core.
+        /// </summary>
+        public ProtoCore.Core LiveRunnerCore
+        {
+            get
+            {
+                return liveRunnerServices.Core;
+            }
+        }
+
+        /// <summary>
+        /// Get DesignScript runtime core.
+        /// </summary>
+        public ProtoCore.RuntimeCore LiveRunnerRuntimeCore
+        {
+            get
+            {
+                return liveRunnerServices.RuntimeCore;
+            }
+        }
+
+
+        /// <summary>
+        /// Return libary service instance.
+        /// </summary>
+        public LibraryServices LibraryServices
+        {
+            get { return libraryServices; }
+        }
+
+        public CodeCompletionServices CodeCompletionServices
+        {
+            get { return codeCompletionServices; }
+        }
+
+        /// <summary>
+        /// A property defining whether the EngineController has been disposed or not.
+        /// This is a conservative field, as there should only be one owner of a valid
```

Comment:
Conservative in the sense that it falsely reports true when it hasn't been disposed?